cmd/reva/app-tokens-create.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"io"
+	"reflect"
+	"strings"
+	"time"
+
+	authapp "github.com/cs3org/go-cs3apis/cs3/auth/applications/v1beta1"
+	authpb "github.com/cs3org/go-cs3apis/cs3/auth/provider/v1beta1"
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	share "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	types "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/auth/scope"
+	"github.com/cs3org/reva/pkg/errtypes"
+)
+
+type appTokenCreateOpts struct {
+	Expiration string
+	Label      string
+	Path       stringSlice
+	Share      stringSlice
+	Unlimited  bool
+}
+
+type stringSlice []string
+
+func (ss *stringSlice) Set(value string) error {
+	*ss = append(*ss, value)
+	return nil
+}
+
+func (ss *stringSlice) String() string {
+	return strings.Join([]string(*ss), ",")
+}
+
+var createOpts *appTokenCreateOpts = &appTokenCreateOpts{}
+
+const layoutTime = "2006-01-02"
+
+func appTokensCreateCommand() *command {
+	cmd := newCommand("token-create")
+	cmd.Description = func() string { return "create a new application tokens" }
+	cmd.Usage = func() string { return "Usage: token-create" }
+
+	cmd.StringVar(&createOpts.Label, "label", "", "set a label")
+	cmd.StringVar(&createOpts.Expiration, "expiration", "", "set expiration time (format <yyyy-mm-dd>)")
+	// TODO(gmgigi96): add support for multiple paths and shares for the same token
+	cmd.Var(&createOpts.Path, "path", "create a token for a file (format path:[r|w]). It is possible specify multiple times this flag")
+	cmd.Var(&createOpts.Share, "share", "create a token for a share (format shareid:[r|w]). It is possible specify multiple times this flags")
+	cmd.BoolVar(&createOpts.Unlimited, "all", false, "create a token with an unlimited scope")
+
+	cmd.ResetFlags = func() {
+		s := reflect.ValueOf(createOpts).Elem()
```

Comment:
Just use `createOpts = &appTokenCreateOpts{}`?

Reply:
In the current implementation, if you do `createOpts = &appTokenCreateOpts{}`, you should change the address of the field in the struct, breaking the pointer in the flags cmds